lightning/src/routing/router.rs

```diff
@@ -1432,6 +1491,27 @@ where L::Target: Logger {
 			return Err(LightningError{err: "Cannot route when there are no outbound routes away from us".to_owned(), action: ErrorAction::IgnoreError});
 		}
 	}
+	// Marshall blinded route hints
+	let mut blinded_route_hints = Vec::with_capacity(payment_params.payee.blinded_route_hints().len());
+	const DUMMY_BLINDED_SCID: u64 = 42;
+	for (idx, (blinded_payinfo, blinded_path)) in payment_params.payee.blinded_route_hints().iter().enumerate() {
+		if blinded_path.blinded_hops.len() == 1 || &blinded_path.introduction_node_id == our_node_pubkey {
+			// If the introduction node is the destination, this hint is for a public node and we can just
+			// use the network graph
+			continue
+		}
+		blinded_route_hints.push(RouteHint(vec![RouteHintHop {
+			src_node_id: blinded_path.introduction_node_id,
+			short_channel_id: DUMMY_BLINDED_SCID + idx as u64,
+			fees: RoutingFees {
+				base_msat: blinded_payinfo.fee_base_msat,
+				proportional_millionths: blinded_payinfo.fee_proportional_millionths,
```

Comment:
Dummy hops don't cost extra fees, since agregated fees for use of the entire blinded path are calculated by the recipient per https://github.com/lightning/bolts/blob/master/proposals/route-blinding.md#blinded-payments, based on the non-dummy hops' feerates. I think the recipient could make the sender overpay, though, if they just added extra fees on top of the aggregated fees?

Reply:
> Dummy hops don't cost extra fees, since agregated fees for use of the entire blinded path are calculated by the recipient per https://github.com/lightning/bolts/blob/master/proposals/route-blinding.md#blinded-payments

Oh I see, thanks

> I think the recipient could make the sender overpay, though, if they just added extra fees on top of the aggregated fees?

I realize reading more of the proposal that recipients are even encouraged to in order to avoid probing, although I also saw [this section](https://github.com/lightning/bolts/blob/master/proposals/route-blinding.md#recipient-pays-fees) where the recipient pays the blinded fees which was interesting. But yea, seems like nothing's really stopping a recipient from getting the sender to overpay as long as its not outrageous...🤷